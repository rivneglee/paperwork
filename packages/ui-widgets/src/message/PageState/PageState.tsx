import React, { FunctionComponent, ReactElement } from 'react';

interface Props {
  image?: string;
  title: string | ReactElement;
  description: string | ReactElement;
  buttons?: ReactElement[];
}

const PageState: FunctionComponent<Props> = ({
  image,
  title,
  description,
  buttons,
}) => (
  <div className="pw-page-state">
    {image && (
      <div className="pw-page-state__image">
        <img src={image} />
      </div>
    )}
    <div className="pw-page-state__context">
      <div className="pw-page-state__title">
        <h3>{title}</h3>
      </div>
      <div className="pw-page-state__description">
        {description}
      </div>
    </div>
    <div className="pw-page-state__action">
      {buttons}
    </div>
  </div>
);

export default PageState;
