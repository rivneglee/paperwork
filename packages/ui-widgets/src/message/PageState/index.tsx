export { default as PageState } from './PageState';
