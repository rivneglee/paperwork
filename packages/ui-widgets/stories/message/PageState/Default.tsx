import React from 'react';

import { BaseTemplate, Button, Card, PageState } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem', display: 'flex', justifyContent: 'center', height: '100hv' }}>
    <BaseTemplate>
      <Card>
        <PageState
          image="https://feelix-assets.s3-ap-southeast-2.amazonaws.com/page-state/empty-state/something-went-wrong.svg"
          title="No result found"
          description="Please change your filter or create some data"
          buttons={[
            <Button color="primary" type="link">Retry</Button>,
            <Button color="primary" type="link">Create some data</Button>,
          ]}
        />
      </Card>
    </BaseTemplate>
  </div>
);
