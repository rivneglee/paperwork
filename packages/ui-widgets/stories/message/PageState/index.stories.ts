import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('PageState', module)
  .add('default', Default);
