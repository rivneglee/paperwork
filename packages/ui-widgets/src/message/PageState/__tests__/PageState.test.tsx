import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PageState from '../PageState';

describe('PageState', () => {
  it('should render properly when has image and buttons', () => {
    const wrapper = shallow(
      <PageState title="Title" description="desc" image="http://foo" buttons={[
        <button/>,
        <button/>,
      ]}/>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render properly without button and image', () => {
    const wrapper = shallow(
      <PageState title="Title" description="desc"/>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
