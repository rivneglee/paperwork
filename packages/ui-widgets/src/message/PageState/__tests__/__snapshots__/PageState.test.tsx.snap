// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageState should render properly when has image and buttons 1`] = `
<div
  className="pw-page-state"
>
  <div
    className="pw-page-state__image"
  >
    <img
      src="http://foo"
    />
  </div>
  <div
    className="pw-page-state__context"
  >
    <div
      className="pw-page-state__title"
    >
      <h3>
        Title
      </h3>
    </div>
    <div
      className="pw-page-state__description"
    >
      desc
    </div>
  </div>
  <div
    className="pw-page-state__action"
  >
    <button />
    <button />
  </div>
</div>
`;

exports[`PageState should render properly without button and image 1`] = `
<div
  className="pw-page-state"
>
  <div
    className="pw-page-state__context"
  >
    <div
      className="pw-page-state__title"
    >
      <h3>
        Title
      </h3>
    </div>
    <div
      className="pw-page-state__description"
    >
      desc
    </div>
  </div>
  <div
    className="pw-page-state__action"
  />
</div>
`;
